[Bug]: Error Update AL Go System File (401 Unauthorized)

### AL-Go version

5.3

### Describe the issue

Hello,

I'm trying to update the AL Go System Files, but I keep getting an error Unauthorized. Do not know the reason, the last time I was able to sync without any problem was a month ago.

Downloading BcContainerHelper latest version from Blob Storage
Import from C:\ProgramData\BcContainerHelper\6.0.22\BcContainerHelper\BcContainerHelper.ps1
BcContainerHelper version 6.0.22
BC.HelperFunctions emits usage statistics telemetry to Microsoft
Running on Windows, PowerShell 5.1.20348.2652
Applying settings from D:\a\D365BC\D365BC\.github\AL-Go-Settings.json
TemplateUrl: https://github.com/microsoft/AL-Go-PTE@main
TemplateSha: 
DownloadLatest: True
The remote server returned an error: ([40](https://github.com/XX/D365BC/actions/runs/10813366223/job/29997227684#step:9:42)1) Unauthorized. 
Unauthorized
...retrying in 1 minute
...retrying in 1 minute
Retry failed as well
The remote server returned an error: (401) Unauthorized. 
Unauthorized
Applying settings from D:\a\D365BC\D365BC\.github\AL-Go-Settings.json
Applying settings from D:\a\D365BC\D365BC\.AL-Go\settings.json
No settings found in D:\a\D365BC\D365BC\.github\Update AL-Go System Files.settings.json
No settings found in D:\a\D365BC\D365BC\.AL-Go\Update AL-Go System Files.settings.json
No settings found in D:\a\D365BC\D365BC\.AL-Go\DesarrolloBC.settings.json
Enabling Microsoft telemetry...
Error: Unexpected error when running action. Error Message: The remote server returned an error: (401) Unauthorized.   Unauthorized , StackTrace: at InvokeWebRequest, D:\a\_actions\microsoft\AL-Go-Actions\v5.3\Github-Helper.psm1: line 112 <- at DownloadTemplateRepository, D:\a\_actions\microsoft\AL-Go-Actions\v5.3\CheckForUpdates\CheckForUpdates.HelperFunctions.ps1: line 31 <- at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\v5.3\CheckForUpdates\CheckForUpdates.ps1: line 72 <- at <ScriptBlock>, D:\a\_temp\904b80da-900a-4996-aa7a-25bc84b1fb26.ps1: line 3 <- at <ScriptBlock>, D:\a\_actions\microsoft\AL-Go-Actions\v5.3\Invoke-AlGoAction.ps1: line 17 <- at <ScriptBlock>, D:\a\_temp\904b80da-900a-4996-aa7a-25bc84b1fb26.ps1: line 2 <- at <ScriptBlock>, <No file>: line 1
Error: Process completed with exit code 1.

### Expected behavior

To update the system files correctly.

### Steps to reproduce

Create a new action, specifying the needed parameters and execute.

### Additional context (logs, screenshots, etc.)

_No response_

--- a/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
+++ b/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
@@ -27,7 +27,7 @@ function DownloadTemplateRepository {
 
     if ($downloadLatest) {
         # Get latest commit SHA from the template repository
-        $templateSha.Value = GetTemplateSha -headers $headers -apiUrl $apiUrl -templateUrl $templateUrl
+        $templateSha.Value = GetLatestTemplateSha -headers $headers -apiUrl $apiUrl -templateUrl $templateUrl
         Write-Host "Latest SHA for $($templateUrl): $($templateSha.Value)"
     }
     $archiveUrl = "$apiUrl/zipball/$($templateSha.Value)"
@@ -41,7 +41,7 @@ function DownloadTemplateRepository {
     return $tempName
 }
 
-function GetTemplateSha {
+function GetLatestTemplateSha {
     Param(
         [hashtable] $headers,
         [string] $apiUrl,

--- a/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
+++ b/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
@@ -20,20 +20,14 @@ function DownloadTemplateRepository {
 
     # Construct API URL
     $apiUrl = $templateUrl.Split('@')[0] -replace "^(https:\/\/github\.com\/)(.*)$", "$ENV:GITHUB_API_URL/repos/`$2"
-    $branch = $templateUrl.Split('@')[1]
 
     Write-Host "TemplateUrl: $templateUrl"
     Write-Host "TemplateSha: $($templateSha.Value)"
     Write-Host "DownloadLatest: $downloadLatest"
 
     if ($downloadLatest) {
-        # Get Branches from template repository
-        $response = InvokeWebRequest -Headers $headers -Uri "$apiUrl/branches?per_page=100" -retry
-        $branchInfo = ($response.content | ConvertFrom-Json) | Where-Object { $_.Name -eq $branch }
-        if (!$branchInfo) {
-            throw "$templateUrl doesn't exist"
-        }
-        $templateSha.Value = $branchInfo.commit.sha
+        # Get latest commit SHA from the template repository
+        $templateSha.Value = GetTemplateSha -headers $headers -apiUrl $apiUrl -templateUrl $templateUrl
         Write-Host "Latest SHA for $($templateUrl): $($templateSha.Value)"
     }
     $archiveUrl = "$apiUrl/zipball/$($templateSha.Value)"
@@ -45,6 +39,34 @@ function DownloadTemplateRepository {
     Expand-7zipArchive -Path "$tempName.zip" -DestinationPath $tempName
     Remove-Item -Path "$tempName.zip"
     return $tempName
+}
+
+function GetTemplateSha {
+    Param(
+        [hashtable] $headers,
+        [string] $apiUrl,
+        [string] $templateUrl
+    )
+    $branch = $templateUrl.Split('@')[1]
+
+    try {
+        $response = InvokeWebRequest -Headers $headers -Uri "$apiUrl/branches?per_page=100" -retry
+        $branchInfo = ($response.content | ConvertFrom-Json) | Where-Object { $_.Name -eq $branch }
+    } catch {
+        if ($_.Exception.Message -like "*401*") {
+            throw "Failed to update AL-Go System Files. Make sure that the personal access token, defined in the secret called GhTokenWorkflow, is not expired and it has permission to update workflows. (Error was $($_.Exception.Message))"
+        } else {
+            throw $_.Exception.Message
+        }
+    }
+
+    if (!$branchInfo) {
+        throw "$templateUrl doesn't exist"
+    }
+
+    return $branchInfo.commit.sha
+    #Write-Host "Latest SHA for $($templateUrl): $($templateSha.Value)"
+    #return $templateSha.Value
 }
 
 function ModifyCICDWorkflow {

--- a/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
+++ b/Actions/CheckForUpdates/CheckForUpdates.HelperFunctions.ps1
@@ -65,8 +65,6 @@ function GetLatestTemplateSha {
     }
 
     return $branchInfo.commit.sha
-    #Write-Host "Latest SHA for $($templateUrl): $($templateSha.Value)"
-    #return $templateSha.Value
 }
 
 function ModifyCICDWorkflow {

